### PR TITLE
Added new lengths for JCB

### PR DESCRIPTION
**Pull request by zeh**
JCB cards were marked as having a length of 16 digits, but they can also be 15 and and 19 digits long. This is not well documented, so it might be a new thing. JCB is a strange credit card.

15 and 16: http://www.regular-expressions.info/creditcard.html
15 and 16: http://developer.ean.com/general-info/valid-card-types/
16-19: https://en.wikipedia.org/wiki/Payment_card_number#Issuer_identification_number_.28IIN.29
16 and 19: https://www.freeformatter.com/credit-card-number-generator-validator.html

### CODE REVIEW

**Comment by crookedneighbor** [comment]
Thanks for opening this PR. 

I believe the 15 length JCB cards only refer to obsolete cards that are no longer in circulation. 

Can you provide more official documentation for this? We'll do some research on our side as well.

**Comment by zeh** [comment]
@crookedneighbor I've tried, but it's hard to find. I've found a lot of second-hand info, always mentioning either 15-16 or 16-19, so you might be right on the 15-digit being outdated.

Now, coming to think of it, I'm not 100% positive it can have a length of 19 digits. The two links above seem to be the only ones mentioning it. [Other](https://github.com/theninja/credit-card-validation/blob/master/src/card.js) [card](https://github.com/stripe/jquery.payment/blob/master/src/jquery.payment.coffee) [validation](https://github.com/juspay/card-validator/blob/master/src/validator.js) [sources](https://github.com/zendframework/zend-validator/blob/master/src/CreditCard.php) seem to always use 15 or 16 digit lengths.

**Comment by zeh** [comment]
@crookedneighbor Do you want me to remove the 15-digit definition? I think you're right in that regard, but haven't been able to find any real info.

(Discover itself only accepts 16-19 digit ones, but they're only one of the companies that seem to carry JCB cards over their network.)

**Comment by crookedneighbor** [comment]
Thanks for the documentation (we don't get notifications for comment edits, so I didn't see it till this morning).

Yes, please remove the 15 digit definition.

As for the 19 digit version, it's possible that it only applies to non-credit card tokenization (such as Apple Pay, Android Pay, etc). We're reviewing it and will get back to you.

File: index.js [hunk 1]
```diff
@@ -121,7 +121,7 @@ types[JCB] = {
   prefixPattern: /^(2|21|213|2131|1|18|180|1800|3|35)$/,
   exactPattern: /^(2131|1800|35)\d*$/,
   gaps: [4, 8, 12],
-  lengths: [16],
+  lengths: [16, 17, 18, 19],
```
Comment (crookedneighbor) [inline index.js]: Our contact at Discover confirmed that JCB cards can be 19 digits. This should be changed to [16, 19].

**Comment by zeh** [comment]
@crookedneighbor Updated, thanks for double checking.

File: index.js [hunk 2]
```diff
@@ -121,7 +121,7 @@ types[JCB] = {
   prefixPattern: /^(2|21|213|2131|1|18|180|1800|3|35)$/,
   exactPattern: /^(2131|1800|35)\d*$/,
   gaps: [4, 8, 12],
-  lengths: [16],
+  lengths: [16, 19],
```
Comment (crookedneighbor) [inline index.js]: I may have been wrong about my last comment. Checking now.

Comment (crookedneighbor) [reply index.js]: Confirmed, you were right. It should be `[16, 17, 18, 19]` after all. My mistake. I misinterpreted what our contact at Discover told us.

File: test/index.js [hunk 3]
```diff
@@ -107,6 +107,7 @@ describe('creditCardType', function () {
       ['18002', 'jcb'],
       ['3530111333300000', 'jcb'],
       ['3566002020360505', 'jcb'],
+      ['3537286818376838569', 'jcb'],
```
Comment (crookedneighbor) [inline test/index.js]: Can we add tests for jcb cards of length 17 and 18 as well?

Comment (zeh) [reply test/index.js]: Is it confirmed that we'll have valid lengths of 17 and 18?

Comment (crookedneighbor) [reply test/index.js]: Yes.

Comment (zeh) [reply test/index.js]: Ah, only saw your previous comment now. Will update!

**Comment by zeh** [comment]
@crookedneighbor Updated, thanks for triple-checking.

**Comment by crookedneighbor** [comment]
We need to do some work on the Braintree backend to allow for this change before we can merge it in. I'll keep you updated, but I expect it to take another week or two before we can merge this in.

**Comment by zeh** [comment]
Fair enough, thanks for the heads up.

File: index.js (hunk 1 above)
Comment (crookedneighbor) [reply index.js]: Our contact with Discover is now giving us conflicting information and saying that it _should_ only be 16 and 19. Trying to get a definitive answer for this.

Comment (zeh) [reply index.js]: Thanks. Keep us posted.

Comment (crookedneighbor) [reply index.js]: Okay, finally got a confirmation. Currently, only 16 or 19 digit cards have been issued, but issuers are allowed to issue 17 and 18 digit cards, so the logic you have is what we should go with for now. 

Should be able to get this merged soon.

Comment (zeh) [reply index.js]: Good to hear!

**Comment by crookedneighbor** [comment]
Released in v6.1.0. Thanks!

**Comment by zeh** [comment]
Thanks!

**Pull request merged**